### PR TITLE
Simplify StreamsWriter a little

### PR DESCRIPTION
**Pull request by alexmalyshev**
Most of the methods on it are equivalent to
RSocketStateMachine::outputFrameOrEnqueue().  So do that.

The main gnarly one that remains is StreamsWriter::writeNewStream(), which does
more than just send a frame.

### CODE REVIEW

File: rsocket/statemachine/RSocketStateMachine.cpp [hunk 1]
```diff
@@ -924,53 +924,20 @@ void RSocketStateMachine::writeNewStream(
   }
 }
 
-void RSocketStateMachine::writeRequestN(StreamId streamId, uint32_t n) {
-  outputFrameOrEnqueue(Frame_REQUEST_N(streamId, n));
+void RSocketStateMachine::writeRequestN(Frame_REQUEST_N&& frame) {
```
Comment (phoad) [inline rsocket/statemachine/RSocketStateMachine.cpp]: Why do we need these specific function? We already call `outputFrameOrEnqueue(std::move(frame))` for each and every one of them. So why don't we just remove them all?

File: rsocket/statemachine/StreamsWriter.h [hunk 2]
```diff
@@ -24,18 +19,12 @@ class StreamsWriter {
       Payload payload,
       bool TEMP_completed) = 0;
 
-  virtual void writeRequestN(StreamId streamId, uint32_t n) = 0;
-
-  virtual void
-  writePayload(StreamId streamId, Payload payload, bool complete) = 0;
+  virtual void writeRequestN(Frame_REQUEST_N&&) = 0;
```
Comment (phoad) [inline rsocket/statemachine/StreamsWriter.h]: Let's simplify it further as renaming this method as `write`?
```
virtual void write(std::unique_ptr<folly::IOBuf> frame)
```
Why do we care about what do we write?

File: rsocket/statemachine/RSocketStateMachine.cpp (hunk 1 above)
Comment (alexmalyshev) [reply rsocket/statemachine/RSocketStateMachine.cpp]: I didn't want to expose `outputFrameOrEnqueue()` just yet, as I'm not sure if that's desirable.  Writing a `REQUEST_N` frame just sends the frame, but writing a `REQUEST_STREAM` frame has to do more bookkeeping within the `RSocketStateMachine`.  It'd be confusing to see why some callers did `outputFrameOrEnqueue(Frame_REQUEST_N{})` and some did `writeNewStream(...)`.

For now I'd rather keep all these methods as separate, to reduce confusion.  I agree that having one function simply "wrapping" another isn't great though.

File: rsocket/statemachine/StreamsWriter.h (hunk 2 above)
Comment (alexmalyshev) [reply rsocket/statemachine/StreamsWriter.h]: See my other comment, but there's different behavior for writing different frames (e.g. `REQUEST_N` versus `REQUEST_STREAM`).

**Pull request merged**